src/main/java/org/embulk/output/mailchimp/MailchimpHttpClient.java

```diff
@@ -0,0 +1,229 @@
+    private static final String API = "https://login.mailchimp.com";
+    private final ObjectMapper jsonMapper = new ObjectMapper()
+            .configure(com.fasterxml.jackson.core.JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS, false)
+            .configure(com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private String accessToken;
+    private Jetty92RetryHelper retryHelper;
+
+    /**
+     * Instantiates a new Mailchimp http client.
+     *
+     * @param task the task
+     */
+    public MailchimpHttpClient(MailchimpOutputPluginDelegate.PluginTask task)
+    {
+        retryHelper = createRetryHelper(task);
+    }
+
+    private JsonNode sendRequest(final String endpoint, final HttpMethod method,
+                                 final MailchimpOutputPluginDelegate.PluginTask task)
+    {
+        return sendRequest(endpoint, method, "", task);
+    }
+
+    private JsonNode sendRequest(final String endpoint, final HttpMethod method, final String content,
+                                 final MailchimpOutputPluginDelegate.PluginTask task)
+    {
+        final String authorizationHeader = getAuthorizationHeader(task);
+
+        try {
+            String responseBody = retryHelper.requestWithRetry(
+                    new StringJetty92ResponseEntityReader(task.getTimeoutMills()),
+                    new Jetty92SingleRequester()
+                    {
+                        @Override
+                        public void requestOnce(HttpClient client, Response.Listener responseListener)
+                        {
+                            Request request = client
+                                    .newRequest(endpoint)
+                                    .accept("application/json")
+                                    .method(method);
+                            if (method == HttpMethod.POST || method == HttpMethod.PUT) {
+                                request.content(new StringContentProvider(content), "application/json");
+                            }
+
+                            if (!authorizationHeader.isEmpty()) {
+                                request.header("Authorization", authorizationHeader);
+                            }
+                            request.send(responseListener);
+                        }
+
+                        @Override
+                        public boolean isResponseStatusToRetry(Response response)
+                        {
+                            int status = response.getStatus();
+                            return status == 429 || status / 100 != 4;
+                        }
+                    });
+
+            return responseBody != null && !responseBody.isEmpty() ? parseJson(responseBody) : MissingNode.getInstance();
+        } catch (HttpResponseException ex) {
+            LOG.error("Exception occurred while sending request: {}", ex.getMessage());
```

Comment:
Ah this is output plugin. So preview isn't a matter.
But I think it's better to throw ConfigException or its sub class to avoid unnecessary retry.